<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <arg name="ns" default="scarab40" />
  <arg name="tf_prefix" default="scarab40" />
  
  <arg name="map_frame"  default="map"/>
  <arg name="odom_frame" default="odom"/>
  <arg name="base_frame" default="base_footprint"/>
  
  <arg name="map_topic"  default="map"/>
  <arg name="odom_topic" default="odom"/>
  <arg name="scan_topic" default="scan" />
  <arg name="pose_topic" default="pose" />
  <arg name="goal_topic" default="move_base_simple/goal"/>

  <arg name="use_hfn" default="true" />




  
<group ns="$(arg ns)">

  <!-- goal_to_action -->
  <!-- topics publised: /move/cancel; /move/feedback; /move/goal; /move/result; /move/status -->
  <node name="goal_to_action" pkg="mrsl_hfn" type="goal_to_action.py">
    <remap from="goal" to="$(arg goal_topic)" />
  </node>

  
  <!-- HFN -->
  <group if="$(arg use_hfn)">
    <node name="pose_stamped" pkg="mrsl_hfn" type="tf_posestamped_node.py" >
      <param name="base_frame_id" value="$(arg tf_prefix)/$(arg base_frame)" />
      <param name="map_frame_id" value="$(arg tf_prefix)/$(arg map_frame)" />
      <remap from="pose" to="$(arg pose_topic)" />
    </node>
    <!-- TOPICS SUBSCRIBED: /pose; /map; /scan; /odom; /move/... -->
    <!-- TOPICS PUBLISED: /path; /marker; /cmd_vel; /inflated_scan; /costmap -->
    <node name="navigation" pkg="mrsl_hfn" type="hfn" respawn="false" output="screen" launch-prefix="" >
      <param name="base_frame_id" value="$(arg tf_prefix)/$(arg base_frame)" />
      <param name="map_frame_id" value="$(arg tf_prefix)/$(arg map_frame)" />

      <param name="cost_occ_prob" value="0.3" />
      <param name="cost_occ_dist" value="0.3" />
      <param name="max_occ_dist" value="1.0" />
      <param name="lethal_occ_dist" value="0.2" />
      <param name="stuck_start" value="50" />
      <!-- 
      <param name="v_opt" value="2.16" />
      <param name="w_max" value="5" />
      <param name="axle_width" value="0.255" />
      <param name="robot_radius" value="0.3" />
      <param name="safety_margin" value="0.1" />
      <param name="social_margin" value="0.2" />
      -->
      
      <remap from="map" to="$(arg map_topic)" />
      <remap from="odom" to="$(arg odom_topic)" />
      <remap from="scan" to="$(arg scan_topic)" />
      <remap from="pose" to="$(arg pose_topic)" />

      <remap from="cmd_vel" to="mobile_base/commands/velocity" />
    </node>  
  </group>
  
  
  <!-- Move Base -->
  <group unless="$(arg use_hfn)">
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <rosparam file="$(find erl_models)/urdf/scarab/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/dwa_local_planner_params.yaml" command="load" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/move_base_params.yaml" command="load" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/global_planner_params.yaml" command="load" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/navfn_global_planner_params.yaml" command="load" />
      <rosparam file="$(find erl_models)/urdf/scarab/param/dummy.yaml" command="load" />
      
      <param name="global_costmap/global_frame" value="$(arg tf_prefix)/$(arg map_frame)"/>
      <param name="global_costmap/robot_base_frame" value="$(arg tf_prefix)/$(arg base_frame)"/>
      <param name="local_costmap/global_frame" value="$(arg tf_prefix)/$(arg odom_frame)"/>
      <param name="local_costmap/robot_base_frame" value="$(arg tf_prefix)/$(arg base_frame)"/>
      <param name="DWAPlannerROS/global_frame_id" value="$(arg tf_prefix)/$(arg odom_frame)"/>

      <remap from="cmd_vel" to="mobile_base/commands/velocity"/>
      <remap from="odom" to="$(arg odom_topic)"/>
      <remap from="scan" to="$(arg scan_topic)"/>    
    </node>
  </group>

</group>

        
</launch>
